interfaces/builtin/lxd_client.go

```diff
@@ -0,0 +1,79 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2016 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+import (
+	"github.com/snapcore/snapd/interfaces"
+)
+
+const lxdClientConnectedPlugAppArmor = `
+# Description: Can access commands and socket from the 'lxd' snap.
+/var/snap/lxd/common/lxd/unix.socket rw,
+`
+
+type LxdClientInterface struct{}
+
+func (iface *LxdClientInterface) Name() string {
+	return "lxd-client"
+}
+
+func (iface *LxdClientInterface) PermanentPlugSnippet(plug *interfaces.Plug, securitySystem interfaces.SecuritySystem) ([]byte, error) {
+	return nil, nil
+}
+
+func (iface *LxdClientInterface) ConnectedPlugSnippet(plug *interfaces.Plug, slot *interfaces.Slot, securitySystem interfaces.SecuritySystem) ([]byte, error) {
+	switch securitySystem {
+	case interfaces.SecurityAppArmor:
+		return []byte(lxdClientConnectedPlugAppArmor), nil
+	case interfaces.SecuritySecComp:
+		return nil, nil
+	}
+	return nil, nil
+}
```

Comment:
From a basic netcat, we see:
```
socket(AF_LOCAL, SOCK_STREAM, 0)        = 3
fcntl(3, F_SETFD, FD_CLOEXEC)           = 0
connect(3, {sa_family=AF_LOCAL, sun_path="/var/lib/lxd/unix.socket"}, 26) = 0
poll([{fd=3, events=POLLIN}, {fd=0, events=POLLIN}], 2, -1
) = 1 ([{fd=0, revents=POLLIN}])
read(0, "\n", 2048)                     = 1
write(3, "\n", 1)                       = 1
poll([{fd=3, events=POLLIN}, {fd=0, events=POLLIN}], 2, -1) = 1 ([{fd=3, revents=POLLIN|POLLHUP}])
read(3, "HTTP/1.1 400 Bad Request\r\nConten"..., 2048) = 88
write(1, "HTTP/1.1 400 Bad Request\r\nConten"..., 88HTTP/1.1 400 Bad Request
Content-Type: text/plain
Connection: close

400 Bad Request) = 88
poll([{fd=3, events=POLLIN}, {fd=0, events=POLLIN}], 2, -1) = 1 ([{fd=3, revents=POLLIN|POLLHUP}])
read(3, "", 2048)                       = 0
shutdown(3, SHUT_RD)                    = 0
close(3)                                = 0
```

So that's:
 - socket (included in base)
 - fcntl (included in base)
 - connect (included in base)
 - poll (included in base)
 - read (included in base)
 - write (included in base)
 - shutdown 
 - close (included in base)

So looks like all that would be allowed for a basic unix socket connection would be "shutdown". Everything else being allowed already by default.

That obviously won't actually let you interact with the socket from something that's written in Go and does extra fancy socket setup, but that's specific to the client rather than the interface.